pkg/availabilitychecks/retry.go

```diff
@@ -0,0 +1,32 @@
+package availabilitychecks
+
+import (
+	"math/rand"
+	"time"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+type stop struct {
+	error
+}
+
+func retry(attempts int, sleep time.Duration, f func() error) error {
```

Comment:
Up to you if you want to adopt it or not, not going to hold up a PR for it, but I use `backoff` [0] for a similar retry/backoff-with-jitter pattern for the upgradeconfigmanager and it's quite nice.
https://github.com/jpillora/backoff

Reply:
i would like to revist this during a refactoring card.